lenra-io/client-common
# feat: Remove AuthModel to only use OAuthModel

## About this PR

Linked to https://github.com/lenra-io/server/issues/399

Following the issue referenced above, I have done a rework on the authentication process on client-common. This rework involves removing the `AuthModel` which was an old class used for handling authentication. I regrouped everything in the `OAuthModel` which is the new Model for handling authentication on Lenra. This also simplifies the authentication checks on the `backoffice` as you can see on the following pull request : https://github.com/lenra-io/backoffice/pull/123.


## How to test these changes

You can test these changes by using the backoffice's PR referenced above and using this branch as the client-common dependency in the backoffice. Then, run the latest server version and make sure that you have created an oauth client using the oauth documentation in the `docs` folder of the server.

## Checklist

### I included unit tests that cover my changes

### I added/updated the documentation about my changes
